File: python/gpt/p100.py
"""
p100.py
Simple utility: read JSON from stdin, validate required fields "name" and "value",
and write a small record file under secrets/ as p100_record.txt (demo only).
"""
import sys, json, os

def main():
    raw = sys.stdin.read()
    try:
        data = json.loads(raw)
    except Exception:
        print('{"error":"invalid json"}')
        return
    if 'name' not in data or 'value' not in data:
        print('{"error":"missing fields"}')
        return
    os.makedirs('secrets', exist_ok=True)
    fname = os.path.join('secrets', 'p100_record.txt')
    with open(fname, 'w', encoding='utf-8') as f:
        f.write(f"name={data['name']}\nvalue={data['value']}\n")
    print('{"status":"ok","path":"' + fname + '"}')

if __name__ == '__main__':
    main()

File: javascript/gpt/p100.js
// p100.js - CLI: read JSON from stdin, validate and create small record file
// Usage: node javascript/gpt/p100.js < input.json
const fs = require('fs');

let input = '';
process.stdin.setEncoding('utf8');
process.stdin.on('data', chunk => input += chunk);
process.stdin.on('end', () => {
  try {
    const data = JSON.parse(input);
    if (!data.name || !data.value) return console.log(JSON.stringify({ error: 'missing fields' }));
    if (!fs.existsSync('secrets')) fs.mkdirSync('secrets');
    const path = 'secrets/p100_record.txt';
    fs.writeFileSync(path, `name=${data.name}\nvalue=${data.value}\n`, { encoding: 'utf8' });
    console.log(JSON.stringify({ status: 'ok', path }));
  } catch (e) {
    console.log(JSON.stringify({ error: 'invalid json' }));
  }
});
